### PR TITLE
WIP: statistics: add AnalysisPriorityQueueV2

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/55906

Problem Summary:

### What changed and how does it work?

TBD, I will update it later.

### Check List

Tests 

- [x] Unit test(TBD)
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+					tableMeta,
+					stats,
+					partitionStats,
+					autoAnalyzeRatio,
+					currentTs,
+				)
+			} else {
+				indicators.ChangePercentage = CalculateChangePercentage(stats, autoAnalyzeRatio)
+				// TODO: check if this is correct.
+				indicators.TableSize = float64(stats.ColNum() * int(stats.RealtimeCount))
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go:364]: Ref https://github.com/pingcap/tidb/pull/56036

File: pkg/statistics/handle/autoanalyze/priorityqueue/build.go [hunk 2]
```diff
@@ -0,0 +1,372 @@
+// Copyright 2024 PingCAP, Inc.
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/build.go:1]: ref https://github.com/pingcap/tidb/pull/56073

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+		changePercent := jobFactory.CalculateChangePercentage(stats)
+		if changePercent == 0 {
+			return nil
+		}
+
+		var job AnalysisJob
+		pruneMode := variable.PartitionPruneMode(sctx.GetSessionVars().PartitionPruneMode.Load())
+
+		is := sctx.GetDomainInfoSchema().(infoschema.InfoSchema)
+		schemaName, ok := is.SchemaNameByTableID(stats.PhysicalID)
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go:286]: ref: https://github.com/pingcap/tidb/pull/56156

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+	statslogutil "github.com/pingcap/tidb/pkg/statistics/handle/logutil"
+	statstypes "github.com/pingcap/tidb/pkg/statistics/handle/types"
+	statsutil "github.com/pingcap/tidb/pkg/statistics/handle/util"
+	"github.com/pingcap/tidb/pkg/util"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// AnalysisPriorityQueueV2 is a priority queue for TableAnalysisJobs.
+type AnalysisPriorityQueueV2 struct {
```
Comment (winoros) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go:40]: Align the name with `Analyze` instead

Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go:40]: The word `analyze` is a verb. I think we need to use `non` here.

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+				job = jobFactory.CreateDynamicPartitionedTableAnalysisJob(
+					schemaName.O,
+					tableMeta,
+					stats,
+					partitionStats,
+				)
+			} else {
+				indicators.ChangePercentage = jobFactory.CalculateChangePercentage(stats)
+				indicators.TableSize = jobFactory.CalculateTableSize(stats)
+				job.SetIndicators(indicators)
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go:373]: ref https://github.com/pingcap/tidb/pull/56248

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue.go [hunk 6]
```diff
@@ -14,7 +14,138 @@
...
+			zap.Float64("weight", weight),
+			zap.Stringer("job", job),
+		)
+	}
+	job.SetWeight(weight)
+	return queue.Push(job)
+}
+
+// BuildAnalysisJobs builds analysis jobs for all eligible tables and partitions.
+func BuildAnalysisJobs(
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue.go:55]: ref https://github.com/pingcap/tidb/pull/56251

File: pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+			statslogutil.StatsLogger().Info("refreshing time")
+			pq.refreshTime()
+		}
+	}
+}
+
+func (pq *AnalysisPriorityQueueV2) fetchDMLUpdate() {
+	if err := statsutil.CallWithSCtx(pq.statsHandle.SPool(), func(sctx sessionctx.Context) error {
+		parameters := exec.GetAutoAnalyzeParameters(sctx)
+		if err := pq.setAutoAnalysisTimeWindow(parameters); err != nil {
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/queue2.go:139]: ref: https://github.com/pingcap/tidb/pull/56335

**Pull request closed without merging**